app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,346 @@
+ *
+ * @param errorType [SyncedTabsView.ErrorType] to be displayed after mapping to a string resource.
+ * @param navController [NavController] to call [NavGraphDirections.actionGlobalTurnOnSync()].
+ */
+@Composable
+fun SyncedTabsErrorItem(errorType: SyncedTabsView.ErrorType, navController: NavController? = null) {
+    val width: Float
+    val dashLength: Float
+    with(LocalDensity.current) {
+        width = 2.dp.toPx()
+        dashLength = 4.dp.toPx()
+    }
+    // As of December 2021, there is not an official way to get a rounded rectangle border with
+    // dashes using Compose. This is a next-best attempt with hard corners.
+    val stroke = Stroke(
+        width = width,
+        pathEffect = PathEffect.dashPathEffect(floatArrayOf(dashLength, dashLength), 0f)
+    )
+    val dashColor = FirefoxTheme.colors.borderDivider
+    Box(
+        Modifier
+            .padding(20.dp)
+            .height(IntrinsicSize.Min)
+    ) {
+        Canvas(Modifier.fillMaxSize()) {
+            drawRoundRect(color = dashColor, style = stroke)
+        }
+        Column(
+            modifier = Modifier
+                .padding(16.dp)
+                .fillMaxWidth()
+        ) {
+            PrimaryText(
+                text = stringResource(id = errorType.toStringRes()),
+                fontSize = 14.sp,
+                modifier = Modifier
+                    .padding(0.dp, 4.dp, 0.dp, 0.dp)
+                    .fillMaxWidth()
+            )
+            if (navController != null) {
+                Spacer(modifier = Modifier.height(8.dp))
+                Row(
+                    modifier = Modifier
+                        .clip(RoundedCornerShape(4.dp))
+                        .background(FirefoxTheme.colors.actionPrimary)
+                        .padding(8.dp, 4.dp)
+                        .clickable(
+                            onClickLabel = stringResource(id = R.string.synced_tabs_sign_in_button),
+                            onClick = {
+                                navController.navigate(NavGraphDirections.actionGlobalTurnOnSync())
+                            }
+                        )
+                ) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.ic_sign_in),
+                        contentDescription = null,
+                        tint = FirefoxTheme.colors.textInverted,
+                    )
+                    Spacer(modifier = Modifier.width(8.dp))
+                    Text(
+                        text = stringResource(id = R.string.synced_tabs_sign_in_button),
+                        color = FirefoxTheme.colors.textInverted,
+                        fontSize = 14.sp,
+                        fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+                        maxLines = 2,
+                        modifier = Modifier.align(Alignment.CenterVertically)
+                    )
```

Comment:
```suggestion
                    Icon(
                        painter = painterResource(id = R.drawable.ic_sign_in),
                        contentDescription = null,
                        tint = FirefoxTheme.colors.textInverted,
                    )
                    
                    Spacer(modifier = Modifier.width(8.dp))
                    
                    Text(
                        text = stringResource(id = R.string.synced_tabs_sign_in_button),
                        color = FirefoxTheme.colors.textInverted,
                        fontSize = 14.sp,
                        fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
                        maxLines = 2,
                        modifier = Modifier.align(Alignment.CenterVertically)
                    )
```

One theme here that I persuade you on is to use new lines to separate these compose components and logical blocks of code. This just helps create a visual separation when reading the code to know when a thing ends.

Reply:
I would also consider looking at the rest file and adding some new lines where it makes sense to not have everything so squished together.

Reply:
I don't think I was even cognizant of doing this in the Compose code haha. Thanks for bringing attention to this; I'll definitely do this going forward and fix it here.